src/GPU3D_Soft.cpp

```diff
@@ -2130,28 +2138,33 @@ void RenderThreadFunc()
         Platform::Semaphore_Wait(Sema_RenderStart);
         if (!RenderThreadRunning) return;
 
-        RenderThreadRendering = true;
-        if (FrameIdentical)
-        {
-            Platform::Semaphore_Post(Sema_ScanlineCount, 192);
-        }
-        else
+        if (!FrameIdentical)
         {
+            // We cannot rely on each post of the previous frame being matched by a wait, because sleep mode can interrupt a frame.
```

Comment:
that's already accounted for via https://github.com/Arisotura/melonDS/blob/master/src/NDS.cpp#L1166

Reply:
Oh, that's new. I'll take a look at that, and getting the 3D buffer to re-render instead of saving the buffer itself.